### DISCUSSION TITLE
File system build error

### DISCUSSION THREAD

**Discussion by kagjerde** [PlummersSoftwareLLC/NightDriverStrip#239] [Q&A]
I have followed the instructions to build the file system [README](https://github.com/PlummersSoftwareLLC/NightDriverStrip/blob/main/site/README.md), but I get the following error:

> $ cd site/
> 
> $ npm install
> 
> up to date, audited 93 packages in 1s
> 
> 24 packages are looking for funding
>   run `npm fund` for details
> 
> found 0 vulnerabilities
> 
> $ npm run start
> 
> > app@0.1.0 start
> > (npm run rebuild& node ./node_modules/serve/build/main.js ./local)
> 
> > app@0.1.0 rebuild
> > node ./node_modules/fs-watch-exec/index.js run -d ./src 'npm run build'
> 
> Watching C:\Users\XXX\Documents\PlatformIO\Projects\NightDriverStrip-main\site\src
**> ''npm' is not recognized as an internal or external command,
> operable program or batch file.**
> 
> $ npm -v
> 9.6.1
> 
> $ node -v
> v16.16.0
> 
> $ pio --version
> PlatformIO Core, version 6.1.6
 
I am new to building the FS this way, but I'am stuck as it seems like the error is in the source or I am using an incompatible version of the tools. All the CLI commands above is entered in the same PIO terminal window.

**Comment by rbergen** [comment]
You're not following the instructions to build the FS. You're following the instructions to run the website locally on your PC; this is purely for testing while developing the site itself. 

In fact, the site never makes it to an FS on the device, as the website's files index.html, main.jsx and favicon.ico are automatically embedded in ("compiled into", if you will) the firmware image when you execute "pio run".

About the error, I think you're running this in a _PowerShell_ PIO terminal window. My guess is that there is something "UNIX-esque" in the  package.json's definition of the `rebuild` script, that PowerShell chokes on. I think we'd need some assistance from @Louis-Riel to further analyse this.

**Comment by rbergen** [comment]
Removing the single quotes from the `rebuild` script in package.json fixes the error. I've opened #240 to promote this to the main tree.

**Comment by rbergen** [comment]
I've just merged #240. As indicated before, this addressed the issue on my system at least.
